src/backend/joanie/core/utils/course_product_relation.py

```diff
@@ -27,8 +30,9 @@ def get_generated_certificates(course_product_relation):
     Return certificates that were published for a course product relation.
     """
     return Certificate.objects.filter(
+        Q(order__course=course_product_relation.course, order__enrollment__isnull=True)
+        | Q(order__course__isnull=True, order__enrollment__isnull=False),
         order__product=course_product_relation.product,
-        order__course=course_product_relation.course,
         order__certificate__isnull=False,
         order__state=ORDER_STATE_COMPLETED,
```

Comment:
Something is wrong here as you will try to generate certificate for all certificate products.

You have to filter per `enrollment__course_run__course` in case of certificate product